src/main/java/ch/njol/skript/events/EvtPlayerInput.java

```diff
@@ -0,0 +1,124 @@
+package ch.njol.skript.events;
+
+import ch.njol.skript.Skript;
+import ch.njol.skript.lang.Literal;
+import ch.njol.skript.lang.SkriptEvent;
+import ch.njol.skript.lang.SkriptParser.ParseResult;
+import org.bukkit.event.Event;
+import org.bukkit.event.player.PlayerInputEvent;
+import org.jetbrains.annotations.Nullable;
+import org.skriptlang.skript.bukkit.InputKey;
+
+import java.util.Set;
+
+public class EvtPlayerInput extends SkriptEvent {
+
+	static {
+		if (Skript.classExists("org.bukkit.Input")) {
+			Skript.registerEvent("Player Input", EvtPlayerInput.class, PlayerInputEvent.class,
+					"[player] (toggle|toggling|1:press[ing]|2:releas(e|ing)) [of] (%-inputkeys%|[a|any] key)",
```

Comment:
i don't like `on toggle key` as a syntax. also, any reason for separating `toggle` and `toggling`? i think it's clearer, but it's not consistent with the other two options (e.g. why not have `release` and `releasing`)

Reply:
> i don't like `on toggle key` as a syntax.

`on toggle key` specifically or all its variations as well?

> also, any reason for separating toggle and toggling? i think it's clearer, but it's not consistent with the other two options (e.g. why not have release and releasing)

I probably just missed by accident

Reply:
I think `toggle` works better for the second pattern, e.g. `on key toggle` or `on sneak toggle`. Actually, does that conflict with any of the sprinting/sneaking events?

Mainly, I don't think these verbs (singular or plural) work well when followed by key.
`on toggle key` - bad
`on press key` - bad
`on release key` - bad
It might be better to require `of` and `a|an`. That is:
`on toggle of a key`
`on press of any key`
`on release of sprint key`

Reply:
> `on sneak toggle`. Actually, does that conflict with any of the sprinting/sneaking events?

No, `on sneak toggle` isn't valid. `key` is required for input key literals. So it would be `on sneak key toggle`